src/boxes/movie.rs

```diff
@@ -59,45 +60,41 @@ impl Mp4Box for Movie {
             tracks: vec![],
             ..Default::default()
         };
+
         let mut index = start + 8; // skip the first 8 bytes that are headers
+                                   // println!("index: {}, len: {}", index, len);
 
-        while index < len {
+        while index < start + len {
             // the first 8 bytes includes the atom size and its name
             // The size is the entire size of the box, including the size and type header, fields, and all contained boxes.
+
             let mut size = vec![0u8; 4];
             let mut name = vec![0u8; 4];
             reader.seek(SeekFrom::Start(index as u64))?;
             reader.read_exact(&mut size)?;
             reader.read_exact(&mut name)?;
             let size = BigEndian::read_u32(&size) as u64;
             let name = std::str::from_utf8(&name)?;
+
             let name = AtomName::from(name);
+            // println!("Name: {:?}, Size: {:?} and Index: {:?}", name, size, index);
```

Comment:
We shouldn't commit the commented code, please remove them

Reply:
Noted.

Reply:
This has been resolved in previous commits.